src/main/scala/io/github/pauljamescleary/petstore/config/PetStoreConfig.scala

```diff
@@ -14,8 +14,10 @@ object PetStoreConfig {
     * return an error.  This should halt the application from starting up.
     */
   def load[F[_]](implicit E: Effect[F]): F[PetStoreConfig] =
-    loadConfig[PetStoreConfig]("petstore") match {
-      case Right(ok) => E.pure(ok)
-      case Left(e) => E.raiseError(new ConfigReaderException[PetStoreConfig](e))
+    E.delay {
+      loadConfig[PetStoreConfig]("petstore") match {
+        case Right(ok) => ok
+        case Left(e) => throw new ConfigReaderException[PetStoreConfig](e)
```

Comment:
I don't know if this is acceptable or not, in general I see "throw" as a red flag and feel like there is a better way to do it.

Is there a different way we can do this without throw?

Reply:
This would work (I tested this locally)...


```
    E.delay(loadConfig[PetStoreConfig]("petstore")).flatMap {
      case Right(ok) => E.pure(ok)
      case Left(e) => E.raiseError(new ConfigReaderException[PetStoreConfig](e))
    }
```

Reply:
Yes, I agree, while both cases would work, your variant looks better.